### PR TITLE
Added Google SignIn

### PR DESCRIPTION
**Pull request by pavan3008**
# Proposed Changes
Integrated Google SignIn functionality into Smiles by utilizing the dependency found at `https://github.com/google/GoogleSignIn-iOS`.

In addition, added `swiftlint.yml` for linting purposes.

|Before|After|
|---|---|
| Simulator Screenshot - iPhone 14 Pro - 2023-05-24 at 14 35 03|Simulator Screenshot - iPhone 14 Pro - 2023-05-24 at 14 21 30|

# Checklist

- [x]  This has been tested
- [x]  Self-documenting code
- [x]  Code is linted

# Instructions to review

Please review the code.

### CODE REVIEW

File: SmilesMiles/SmilesMilesApp.swift [hunk 1]
```diff
@@ -6,12 +6,21 @@
...
 
 @main
 struct SmilesMilesApp: App {
     var body: some Scene {
         WindowGroup {
             MainView()
+                .onOpenURL { url in
+                              GIDSignIn.sharedInstance.handle(url)
+                            }
+                .onAppear {
```
Comment (KedarDhere) [inline SmilesMiles/SmilesMilesApp.swift:19]: Comment this if not required

**Review by KedarDhere — APPROVED** [review]
Code looks good.

File: SmilesMiles/SmilesMilesApp.swift (hunk 1 above)
Comment (pavan3008) [reply SmilesMiles/SmilesMilesApp.swift:19]: Thanks! Added a print statement to verify the previous sign-in.

**Comment by pavan3008** [comment]
Thanks for reviewing.

**Pull request merged**